modin/backends/pandas/query_compiler.py

```diff
@@ -1267,22 +1267,39 @@ def median(self, **kwargs):
         func = self._build_mapreduce_func(pandas.DataFrame.median, **kwargs)
         return self._full_axis_reduce(axis, func)
 
-    def memory_usage(self, **kwargs):
+    def memory_usage(self, axis=0, **kwargs):
         """Returns the memory usage of each column.
 
         Returns:
             A new QueryCompiler object containing the memory usage of each column.
         """
+        if self._is_transposed:
+            return self.transpose().memory_usage(axis=1, **kwargs)
 
         def memory_usage_builder(df, **kwargs):
-            return df.memory_usage(**kwargs)
+            axis = kwargs.pop("axis")
+            # We have to manually change the orientation of the data within the
+            # partitions because memory_usage does not take in an axis argument
+            # and always does it along columns.
+            if axis:
+                df = df.T
+            result = df.memory_usage(**kwargs)
+            # We add back in the axis kwarg for the sum_memory_usage reduction
+            # function.
+            kwargs["axis"] = axis
```

Comment:
Is this necessary? They are handled in separate tasks, so I can't imagine this is needed.